basic_autonomy/src/basic_autonomy.cpp

```diff
@@ -878,8 +878,34 @@ namespace basic_autonomy
             ROS_DEBUG_STREAM("Corresponding to point: x: " << all_sampling_points[buffer_pt_index].x() << ", y:" << all_sampling_points[buffer_pt_index].y());
 
             if(nearest_pt_index + 1 >= buffer_pt_index){
-                ROS_WARN_STREAM("Current state is at or past the planned end distance. Couldn't generate trajectory");
-                return {};
+                
+                lanelet::BasicPoint2d current_pos(state.x_pos_global, state.y_pos_global);
+                lanelet::BasicPoint2d ending_pos(ending_state.x_pos_global, ending_state.y_pos_global);
+
+                if( nearest_pt_index > buffer_pt_index ||
+                    (wm->routeTrackPos(ending_pos).downtrack -  wm->routeTrackPos(current_pos).downtrack) < epsilon_ ){
```

Comment:
I think this is still equivalent to `nearest_pt_idx >= buffer_pt_idx`, except we're now using an epsilon for the equals side of the comparison. Would it make more sense for this whole conditional to be:

`if (wm->routeTrackPos(ending_pos).downtrack < wm.routeTrackPos(current_pos).downtrack) { ... }`

That should cover all cases where we can't generate a trajectory right?

Reply:
Yes that makes sense, only thing is nearest_pt_index being greater than buffer_pt_index will cause an exception further down in this function on line 913. 
Do we want to have that check happen separately?